packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,739 @@
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree as JS Object (${TestPrimitives[dataType]}): writes with ${i} nodes`,
+					before: async () => {},
+					benchmarkFn: () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Wide, dataType);
+					},
+				});
+			}
+			for (const i of deepTestSizes) {
+				let tree: ISharedTree;
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Deep Tree as JS Object (${TestPrimitives[dataType]}): manipulations with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Deep, dataType);
+					},
+					benchmarkFn: () => {
+						manipulateTreeAsJSObject(tree, dataType, random);
+					},
+				});
+			}
+			for (const i of wideTestSizes) {
+				let tree: ISharedTree;
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree as JS Object (${TestPrimitives[dataType]}): manipulations with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Wide, dataType);
+					},
+					benchmarkFn: () => {
+						manipulateTreeAsJSObject(tree, dataType, random);
+					},
+				});
+			}
+		}
+	});
+	describe("Cursors", () => {
+		for (let dataType = 0 as TestPrimitives; dataType <= 4; dataType++) {
+			for (const i of deepTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Deep Tree (${TestPrimitives[dataType]}) with cursor: reads with ${i} nodes`,
+					before: async () => {
+						tree = await generateTestTree(i, TreeShape.Deep, dataType, random);
+					},
+					benchmarkFn: () => {
+						readCursorTree(tree.forest, i, TreeShape.Deep);
+					},
+				});
+			}
+			for (const i of wideTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree (${TestPrimitives[dataType]}) with cursor: reads with ${i} nodes`,
+					before: async () => {
+						tree = await generateTestTree(i, TreeShape.Wide, dataType, random);
+					},
+					benchmarkFn: () => {
+						readCursorTree(tree.forest, i, TreeShape.Wide);
+					},
+				});
+			}
+			for (const i of deepTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Deep Tree (${TestPrimitives[dataType]}) with cursor: writes ${i} nodes`,
+					before: () => {},
+					benchmarkFn: async () => {
+						tree = await generateTestTree(i, TreeShape.Deep, dataType, random);
+					},
+				});
+			}
+			for (const i of wideTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree (${TestPrimitives[dataType]}) with cursor: writes ${i} nodes`,
+					before: () => {},
+					benchmarkFn: async () => {
+						tree = await generateTestTree(i, TreeShape.Wide, dataType, random);
```

Comment:
Since this says its measuring writes, and not async test tree initialization, you shouldn't do the async test tree initialization inside the benchmark.

Also using different random data inside each iteration of the benchmark is bad for two reasons:
1. Adds noise to be benchmark, which may be different for different runs and for runs you are comparing against since different iterations get different data and not all runs will run the same number of iterations
2. generating random numbers takes time, and you are including that. I'm not sure what you are using them for in this case, or how many, but in general its not a great thing to do.

Reply:
Moved the initialization outside the benchmark, removed the random data generation.